File: JavaScript/isAnagram.js

let s = "rat" 
let t = "car"



var isAnagram = function(s, t) {
   
    let arrS = s.split('')
    let arrT = t.split('')
    let sortS = arrS.sort()
    let sortT = arrT.sort()
    console.log(arrS)
    if (sortS.length != sortT.length) return false
       
     for (let x = 0; x < sortS.length; x++) {
        let charT = sortT[x]
        let charS = sortS[x]
        
            if ( charT != charS) return false  
                
        }
    return true


};


console.log(isAnagram(s, t))

File: Python/isAnagram.py

s = "cat"
t = "rat"



def isAnagram (s, t):

    sort_s =sorted(s)
    sort_t = sorted(t)
    if len(sort_s) != len(sort_t):
        return False
    
    for char_s, char_t in zip(sort_s, sort_t):
        
         if char_s != char_t: return False
    return True

    
print(isAnagram(s, t))    